confluence/server/scio_search/src/main/java/com/askscio/atlassian_plugins/confluence/impl/ScioSearchInfoResponse.java

```diff
@@ -0,0 +1,126 @@
+package com.askscio.atlassian_plugins.confluence.impl;
+
+import static com.askscio.atlassian_plugins.confluence.impl.MyPluginComponentImpl.TARGET_CONFIG_KEY;
+
+import com.askscio.atlassian_plugins.confluence.impl.Utils.PluginStatus;
+import com.atlassian.confluence.setup.settings.SettingsManager;
+import com.atlassian.confluence.util.GeneralUtil;
+import com.atlassian.extras.common.log.Logger;
+import com.atlassian.plugin.Plugin;
+import com.atlassian.plugin.PluginAccessor;
+import com.atlassian.plugin.PluginInformation;
+import com.atlassian.sal.api.pluginsettings.PluginSettings;
+import com.atlassian.sal.api.pluginsettings.PluginSettingsFactory;
+import com.atlassian.sal.api.user.UserManager;
+import com.atlassian.sal.api.user.UserProfile;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import org.codehaus.jackson.annotate.JsonAutoDetect;
+import org.codehaus.jackson.annotate.JsonAutoDetect.Visibility;
+
+@JsonAutoDetect(fieldVisibility = Visibility.ANY)
+public class ScioSearchInfoResponse {
+
+  private static final Logger.Log logger = Logger.getInstance(ScioSearchInfoResponse.class);
+  private final UserInfo userInfo;
+  private final InstanceInfo instanceInfo;
+  private final ScioPluginInfo scioPluginInfo;
+
+  public ScioSearchInfoResponse(UserInfo userInfo, InstanceInfo instanceInfo,
+      ScioPluginInfo scioPluginInfo) {
+    this.userInfo = userInfo;
+    this.instanceInfo = instanceInfo;
+    this.scioPluginInfo = scioPluginInfo;
+  }
+
+  @JsonAutoDetect(fieldVisibility = Visibility.ANY)
+  public static class UserInfo {
+
+    private final String userKey;
+    private final String userName;
+    private final String fullName;
+    private final String email;
+    private final boolean isAdmin;
+
+    public UserInfo(UserManager userManager) {
+      final UserProfile profile = userManager.getRemoteUser();
+      this.userKey = profile.getUserKey().getStringValue();
+      this.userName = profile.getUsername();
+      this.fullName = profile.getFullName();
+      this.email = profile.getEmail();
+      this.isAdmin = Utils.isCurrentUserAdmin(userManager);
+    }
+  }
+
+  @JsonAutoDetect(fieldVisibility = Visibility.ANY)
+  public static class InstanceInfo {
+
+    private final String version;
+    private final String baseUrl;
+    private final List<InstalledPluginInfo> installedPluginInfos;
+
+    public InstanceInfo(SettingsManager settingsManager, PluginAccessor pluginAccessor,
+        UserManager userManager,
+        boolean getInstalledPlugins) {
+      this.version = GeneralUtil.getVersionNumber();
+      this.baseUrl = settingsManager.getGlobalSettings().getBaseUrl();
+      if (getInstalledPlugins && Utils.isCurrentUserAdmin(userManager)) {
+        this.installedPluginInfos = new ArrayList<>();
+        Collection<Plugin> installedPlugins = pluginAccessor.getPlugins();
```

Comment:
This feels a little bit nosy. I assume we want this to fish for plugins that might affect macros?

At minimum we should call this out in the documentation and release notes. But I'd probably prefer just drop it until we really need it.

Reply:
Okay, removing this for now